packages/client/src/utils/hooks/ForecastDataHook.ts

```diff
@@ -0,0 +1,48 @@
+import useRemoteForecastService from './ForecastServiceHook'
+import axios from 'axios'
+
+export interface RegionData {
+  location: string
+  workLoadExecutionTime: number
+}
+
+export interface OptimalTime {
+  region: string
+  location: string
+  timestamp: string
+  duration: number
+  value: number
+}
+export interface DataResult<T> {
+  result: T
+  error: Error | null
+  loading: boolean
+}
+
+export const useForecastData = (
+  params: Map<string, string[]>,
+): DataResult<Map<string, OptimalTime[]>> => {
+  const baseUrl = process.env.REACT_APP_CARBON_AWARE_SDK_BASE_URL
+
+  const promiseArray = []
+  for (const parameter of params) {
+    const promise = axios.get(
+      `${baseUrl}/emissions/forecasts/current?location=${parameter[1][0]}&windowSize=${parameter[1][1]}`,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          'Access-Control-Allow-Origin': '*',
+        },
+      },
+    )
+    promiseArray.push(promise)
+  }
+
+  const { data, error, loading } = useRemoteForecastService(promiseArray)
+  let index = 0
+  const result = new Map()
+  for (const [key] of params) {
+    result.set(key, data[index++])
+  }
+  return { result, error, loading }
+}
```

Comment:
We should use our API as a "middle-ma" to making calls to the Carbon Aware SDK. The client should not have an additional service to rely on besides the CCF API. I'd recommend looking into refactoring this in a way to where SDK calls, configs, and any non-presentational logic occur through the API, and this Data/Service hook (which can be combined into one, not sure if separate ones are needed) only handles making the call to the CCF endpoint.

Since it seems that the forecast logic also relies on emissions data, perhaps the fetching and formatting of the data can occur whenever estimates are initially fetch, and have it be included in the results? Since it relies on service estimates, this may cause problems on the client when dealing with large scale data. Open to discussing